Frontend/ui-common/src/Components/PopUpModal/Controller.purs

```diff
@@ -66,9 +68,18 @@ type Config = {
     tipSelected :: String,
     fareEstimateText :: String,
     tipSelectedText :: String,
-    backgroundColor  :: String
+    backgroundColor  :: String,
+    popUpStatus :: PopUpStatus,
+    buttonClickType :: Int 
```

Comment:
what does buttonClickType means? we will be using it for what purpose?

Reply:
From popUpModal there are so many action which are being triggered (OnDismiss , OnButton1Click, OnButton2Click, OnImageCLick) and on click of all there button we want the first closing animation should execute then these action so to  know which action triggered closing animation, we are using this buttonCLickType.

Reply:
Then it's not button click, we can call it Action type maybe, as OnDismiss is also not a button